nemo/collections/nlp/data/dialogue/data_processor/design_data_processor.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+# Copyright 2019 The Google Research Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import pandas as pd
+
+from nemo.collections.nlp.data.dialogue.data_processor.data_processor import DialogueDataProcessor
+from nemo.collections.nlp.data.dialogue.input_example.input_example import DialogueInputExample
+
+__all__ = ['DialogueDesignDataProcessor']
+
+
+class DialogueDesignDataProcessor(DialogueDataProcessor):
+    """Data Processor for Design Dataset"""
+
+    def __init__(self, data_dir: str, tokenizer: object):
+        """
+        Constructs DialogueDesignDataProcessor
+        Args:
+            data_dir: path to data directory
+            tokenizer: tokenizer object
+        """
+        self.data_dir = data_dir
+        self._tokenizer = tokenizer
+
+    def open_csv(self, filename):
+        """
+        Reads file into a list
+        """
+        filename = os.path.join(self.data_dir, filename)
+        with open(filename, "r", encoding="UTF-8") as f:
+            df = pd.read_csv(filename)
+        return df.to_dict(orient='index')
+
+    def get_dialog_examples(self, dataset_split: str):
+        """
+        Process raw files into DialogueInputExample
+        Args: 
+            dataset_split: {train, dev, test}
+        Train contains 90% of the dataset, Dev is the other 10% 
+        Test contains the whole dataset (Dev + Train)
```

Comment:
This is done as this Design dataset (as well as Mellon QA dataset) is small (~100) and primarily used in a zero shot setting, so this allows evaluation to be done on the entire dataset. Technically, we might not need to have a `train` or `dev` set for these datasets, but it might be good to keep them here for possible future few shot models.

Agree on controlling percentage of data!

Reply:
For the general comments, I have addressed the points on having more complete docstrings coverage as well as logging over print. As for the point on overlap between the `dialogue` module and modules like `zero_shot_intent`, we might want to have a discussion on whether to continue support standalone models `zero_shot_intent` in the long run (in the short run, this is probably necessary as they are likely to be deployed by various users). One argument for not supporting the standalone models further is that they are less extensible in terms of modelling (dialogue.zero_shot_intent_model currently supports a BART based MNLI model as well) and loading various tasks (as they would need additional scripts to preprocess a dataset, which I think is less maintainable than DataProcessor classes). On the other hand, it might be simpler to use from a deployment POV as it's more of a single purpose model, which much less configuration for users.